src/go/rpk/pkg/raid0/setup.go

```diff
@@ -0,0 +1,391 @@
+			}
+		} else {
+			rr.l.Warnf("skipping blkdiscard on %q: %s", dpath, err)
+		}
+
+		wipeFS, err := shelltool.WipeFS(dpath).All().Build(ctx)
+		if err != nil {
+			return fmt.Errorf("preparing wipefs command: %w", err)
+		}
+
+		rr.l.Debugf("running %q...", wipeFS.String())
+		if out, err := wipeFS.CombinedOutput(); err != nil {
+			return fmt.Errorf("%s: %w", string(out), err)
+		}
+	}
+
+	return nil
+}
+
+// create invokes mdadm(8) to create a RAIDO device and removes all previous filesystem
+// signatures from the RAID0 device.
+func (rr *RedpandaRAID) create(ctx context.Context) error {
+	mdadmCreate, err := shelltool.MdadmCreate(rr.devicePath).
+		Verbose().
+		Force().
+		Run().
+		// So the RAID is not re-assembled as foreign, after host migrations in
+		// cloud environments.
+		HomeHost("any").
+		Level("0").
+		ChunkSize(int(rr.chunkSizeKiB)).
+		DeviceNumber(len(rr.memberDevicePaths)).
+		Devices(rr.memberDevicePaths...).
+		Build(ctx)
+	if err != nil {
+		return err
+	}
+
+	rr.l.Debugf("running %q...", mdadmCreate.String())
+	if out, err := mdadmCreate.CombinedOutput(); err != nil {
+		return fmt.Errorf("%s: %w", string(out), err)
+	}
+
+	// since wipefs -a on individual disks does not clear potential previous signatures
+	// present on the RAID device.
+	wipeFS, err := shelltool.WipeFS(rr.devicePath).All().Build(ctx)
+	if err != nil {
+		return err
+	}
+
+	rr.l.Debugf("running %q...", wipeFS.String())
+	if out, err := wipeFS.CombinedOutput(); err != nil {
+		return fmt.Errorf("%s: %w", string(out), err)
+	}
+
+	rr.l.Debugf("updating mdadm.conf...")
+	if err := rr.updateMDADMConf(ctx); err != nil {
+		return fmt.Errorf("updating mdadm.conf file: %w", err)
+	}
+
+	// Updates initramfs' mdadm.conf, to make sure the RAID device can be
+	// correctly re-assembled at booting time.
+	rr.l.Debugf("updating initramfs...")
+	if err := rr.updateInitRAMFS(ctx); err != nil {
+		return fmt.Errorf("updating initramfs: %w", err)
+	}
+
+	return nil
+}
+
+// format runs mkfs.xfs on the RAID device, as quick as possible.
+func (rr *RedpandaRAID) format(ctx context.Context) error {
+	makeXFS, err := shelltool.MakeXFS(rr.devicePath).
+		BlockSize(rr.blockSizeB).
+		NoDiscard().
+		Build(ctx)
+	if err != nil {
+		return err
+	}
+
+	rr.l.Debugf("running %q...", makeXFS.String())
+	if out, err := makeXFS.CombinedOutput(); err != nil {
+		return fmt.Errorf("%q: %w: %s", makeXFS.String(), err, string(out))
+	}
+
+	return nil
+}
+
+func (rr *RedpandaRAID) mount(ctx context.Context) error {
+	blockID, err := shelltool.BlockID(rr.devicePath).
+		MatchTag("UUID").
+		OutputFormat("value").Build(ctx)
+	if err != nil {
+		return err
+	}
+
+	rr.l.Debugf("running %q...", blockID.String())
+	output, err := blockID.Output()
+	if err != nil {
+		return fmt.Errorf("%q: %w", blockID.String(), err)
+	}
+
+	uuid := strings.TrimSpace(string(output))
+
+	tpl, err := template.ParseFS(mountUnitTpl, "systemd/var-lib-redpanda.mount")
```

Comment:
suggestion: I feel the mount path should be configurable, shouldn't it? Or do we always require the redpanda data dir to be here?

Reply:
It is configurable, this is only the embedded template for the systemd unit.